FirstClasses/fibonacci_sum/src/fibonacci_sum.erl

```diff
@@ -6,5 +6,17 @@ fibonacci_sum() ->
     Limit = 4 * 1000 * 1000,
     fibonacci_sum(1, 1, 0, Limit).
 
-fibonacci_sum(_Prev, _Curr, _Sum, _Limit) ->
-    1.
+fibonacci_sum(_Prev, Curr, Sum, Limit) ->
+    fibonacci_sum(Curr, Sum, Limit-1).
+
+fibonacci_sum(_, Sum, Limit) when Sum >= Limit -> 0;
+fibonacci_sum(Iter, _, Limit) ->
+    V = fibonacci(Iter),
+    case V rem 2 of
+        0 -> fibonacci(Iter) + fibonacci_sum(Iter + 1, V,  Limit - 1);
+        1 -> fibonacci_sum(Iter + 1, V, Limit - 1)
+    end.
+
+fibonacci(0) -> 1;
```

Comment:
Fibonacci in the classical definition is actually a very good example of a performance issue for bigger numbers. I prepared the following benchmark to show you what is the issue:
```erlang
-module(m).

-export([compare/1]).

fibonacci(0) -> 1;
fibonacci(1) -> 1;
fibonacci(Num) -> fibonacci(Num - 1) + fibonacci(Num - 2).

tail_fibonacci(N) ->
    tail_fib(N, {1, 1}).

tail_fib(0, {Prev, Curr}) -> Prev;
tail_fib(N, {Prev, Curr}) ->
    tail_fib(N - 1, {Curr, Curr + Prev}).

compare(SampleSize) ->
    {Time, Value} = timer:tc(fun() -> fibonacci(SampleSize) end),
    {TimeRec, Value} = timer:tc(fun() -> tail_fibonacci(SampleSize) end),
    io:fwrite("Classic recursion takes: ~p seconds\n", [Time / (1000 * 1000)]),
    io:fwrite("Tail recursion takes: ~p seconds\n", [TimeRec / (1000 * 1000)]).
```
And these are the results on my mac
```
1> [m:compare(S) || S <- [1, 5, 10, 15, 20, 25, 30, 35, 40]].
Classic recursion takes: 1.0e-6 seconds
Tail recursion takes: 1.0e-6 seconds
Classic recursion takes: 1.0e-6 seconds
Tail recursion takes: 0.0 seconds
Classic recursion takes: 4.0e-6 seconds
Tail recursion takes: 0.0 seconds
Classic recursion takes: 3.5e-5 seconds
Tail recursion takes: 1.0e-6 seconds
Classic recursion takes: 4.03e-4 seconds
Tail recursion takes: 1.0e-6 seconds
Classic recursion takes: 0.006042 seconds
Tail recursion takes: 2.0e-6 seconds
Classic recursion takes: 0.046962 seconds
Tail recursion takes: 1.0e-6 seconds
Classic recursion takes: 0.478911 seconds
Tail recursion takes: 2.0e-6 seconds
Classic recursion takes: 5.328895 seconds
Tail recursion takes: 1.0e-6 seconds
[ok,ok,ok,ok,ok,ok,ok,ok,ok]
```